src/cli/common.rs

```diff
@@ -424,14 +424,36 @@ pub fn list_installed_components(toolchain: &Toolchain<'_>) -> Result<()> {
     Ok(())
 }
 
-pub fn list_toolchains(cfg: &Cfg, is_verbose: bool) -> Result<()> {
-    let toolchains = cfg.list_toolchains()?;
-    let toolchain_info = if is_verbose {
-        format!("\t{}", &cfg.rustup_dir.display())
+fn get_path_delimiter() -> &'static str {
+    #[cfg(windows)]
+    fn get_delimiter() -> &'static str {
+        "\\"
+    }
+    #[cfg(not(windows))]
+    fn get_delimiter() -> &'static str {
+        "/"
+    }
+    get_delimiter()
+}
+
+fn print_toolchain_path(cfg: &Cfg, toolchain: String, if_default: &str, verbose: bool) {
+    let path_delimiter = get_path_delimiter();
+    let toolchain_path = if verbose {
+        format!(
+            "\t{}{}{}{}",
+            cfg.toolchains_dir.display(),
+            path_delimiter,
+            &toolchain,
+            path_delimiter
+        )
```

Comment:
Surely this whole thing would be better done along the lines of:

```rust
let toolchain_path = if verbose {
    let tpath = cfg.toolchains_dir.clone();
    tpath.push(&toolchain);
    format!("\t{}", tpath.display())
}
```

Rather than adding an extra function for the delimiter?